pkg/scaling/scale_handler.go

```diff
@@ -221,13 +221,22 @@ func (h *scaleHandler) checkScaledJobScalers(ctx context.Context, scalers []scal
 		scalerLogger := h.logger.WithValues("Scaler", scaler)
 
 		isTriggerActive, err := scaler.IsActive(ctx)
-		scaler.Close()
+		// scaler.Close() // TODO why does it closes?
```

Comment:
some scalers need to be closed or they will leak connections.

Reply:
Hi @ahmelsayed 
Can we identify, if it requires close it in here?
If you give me the scaler name, then I'll test it as well and compare the behavior.

Reply:
rabbitmq and kafka are the ones that come to mind

Reply:
In case of the rabbit mq, if I keep the scaler.Close() here, it emits error and doesn't work. I'll investigate more and find a place to close or proper condition for closing it. :) 
I have a kafka cluster, so I can try it as well.

Reply:
Hi @ahmelsayed 
I change the place of the `scaler.Close()` if we put it on the original place, following GetMetrics/GetMetricSpecForScaling can't use the connection. So that it cause queue length 0. So that it won't scale. The new place will fine.